### DISCUSSION TITLE
SDM630 Support in 2.7.8

### DISCUSSION THREAD

**Discussion by Messiam** [johanmeijer/grott#278] [Q&A]
Hi there!
Thank you for this nice project

I was curious if the SDM630 data is published to MQTT in 2.7.8 Beta.

My setup is SDM630 Modbus connected to Growatt 10000 tl3-s

Do I have to enable something special in the config?

I only get this under energy/growatt:

`{"device": "EHE3A24XXXX", "time": "2023-02-08T13:14:03", "buffered": "no", "values": {"datalogserial": "JPC6A2XXX", "pvserial": "EHE3A2XXXX", "pvstatus": 1, "pvpowerin": 73842, "pv1voltage": 5567, "pv1current": 66, "pv1watt": 36742, "pv2voltage": 5456, "pv2current": 68, "pv2watt": 37100, "pvpowerout": 68408, "pvfrequentie": 5000, "pvgridvoltage": 4122, "pvgridcurrent": 96, "pvgridpower": 22799, "pvgridvoltage2": 4129, "pvgridcurrent2": 95, "pvgridpower2": 22600, "pvgridvoltage3": 4091, "pvgridcurrent3": 96, "pvgridpower3": 22628, "pvenergytoday": 246, "pvenergytotal": 165946, "totworktime": 48111468, "pvtemperature": 476, "pvipmtemperature": 400, "epv1today": 133, "epv1total": 90647, "epv2today": 130, "epv2total": 87493, "epvtotal": 178140}}`

**Comment by johanmeijer** [comment]
It is standard sent to "energy/meter" topic. You can change the topic with:

mqttmtopicname= "energy/meter"

parameter

**Reply by Messiam** [reply]
Thank you.
This means the Smartmeter values are not received correctly by the inverter ?
Because I can't identify them in the mqtt message.

**Comment by johanmeijer** [comment]
We have to see in the log if there is a data record from the smartmeter. Do you see something about T06501b or T060120 layout in the log?

If you have a shinewifi it can also be T06011b (but probably that is not handled well in the code I see now :( ).

**Comment by Messiam** [comment]
Hi,
it's a ShineWIFI-S Firmware 3.0.0.0

I can't find T06501b or T060120 or T06011b in the log.

Unfortunately I can't even guarantee that my inverter is correctly communicating with the smartmeter because we still have to turn the smartmeter (the electrician installed it wrongly IN/OUT swapped). Just was curios if the values are already present in grott :-)
[grott-log.txt](https://github.com/johanmeijer/grott/files/10695210/grott-log.txt)

**Reply by BarmanMaz** [reply]
> Hi, it's a ShineWIFI-S Firmware 3.0.0.0
> 
> I can't find T06501b or T060120 or T06011b in the log.
> 
> Unfortunately I can't even guarantee that my inverter is correctly communicating with the smartmeter because we still have to turn the smartmeter (the electrician installed it wrongly IN/OUT swapped). Just was curios if the values are already present in grott :-) [grott-log.txt](https://github.com/johanmeijer/grott/files/10695210/grott-log.txt)

Hi, can you please help me suggesting how to have the full log, I tyed with HA bu no way, i have only the mqtt datas, i need to have all the flow.
thanks

**Comment by johanmeijer** [comment]
Yes they will be available in Grott and sent to a different MQTT topic, used in pvoutput and added to influxdb.
